### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by Deren-Liao**
LogsViewerViewModel.cs is mainly for
https://github.com/GoogleCloudPlatform/google-cloud-visualstudio/issues/532 Log streaming may miss log items or add duplicate items

LogsViewerToolWindowControl.xaml is mainly for
https://github.com/GoogleCloudPlatform/google-cloud-visualstudio/issues/359 Logs Viewer, datagrid Message column, auto resize to window width and use TextTrimming="CharacterEllipsis" #359

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs [hunk 1]
```diff
@@ -834,10 +848,21 @@ private void AutoReload()
...
+                    Debug.WriteLine($"_latestLogItem is {_latestLogItem?.TimeStamp}, {_latestLogItem?.Message}");
+                    if (DateTimePickerModel.IsDescendingOrder)
+                    {
+                        DateTimePickerModel.DateTimeUtc = DateTime.UtcNow;
+                    }
 
-                    _filter = ComposeSimpleFilters(ignoreTimeStamp: true)
-                        + String.Format($" timestamp>\"{_lastLogTime.ToString("O")}\" ");
+                    StringBuilder filter = new StringBuilder(ComposeSimpleFilters());
+                    filter.AppendLine($"timestamp<\"{DateTime.UtcNow.AddSeconds(-LogStreamingDelayInSeconds).ToString("O")}\"");
```
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: And here.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs [hunk 2]
```diff
@@ -774,20 +791,17 @@ private string ComposeSimpleFilters(bool ignoreTimeStamp = false)
...
-                    if (DateTimePickerModel.DateTimeUtc < DateTime.UtcNow)
-                    {
-                        filter.AppendLine($"timestamp<=\"{DateTimePickerModel.DateTimeUtc.ToString("O")}\"");
-                    }
-                }
-                else
+                if (DateTimePickerModel.DateTimeUtc < DateTime.UtcNow)
                 {
-                    filter.AppendLine($"timestamp>=\"{DateTimePickerModel.DateTimeUtc.ToString("O")}\"");
+                    filter.AppendLine($"timestamp<=\"{DateTimePickerModel.DateTimeUtc.ToString("O")}\"");
```
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: You could rewrite this line as 
`filter.AppendLine($"timestamp<=\"{DateTimePickerModel.DateTimeUtc:O}\"");

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs [hunk 3]
```diff
@@ -774,20 +791,17 @@ private string ComposeSimpleFilters(bool ignoreTimeStamp = false)
...
                 }
             }
+            else
+            {
+                filter.AppendLine($"timestamp>=\"{DateTimePickerModel.DateTimeUtc.ToString("O")}\"");
```
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: And here.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs (hunk 2 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerViewModel.cs]: Good to learn. This is the part I like code review feedback.  Thank you.

**Comment by Deren-Liao** [comment]
Thank you Jim for doing the review.

**Pull request merged**